cmd/natc/natc.go

```diff
@@ -219,6 +256,35 @@ func (c *connector) handleDNS(pc net.PacketConn, buf []byte, remoteAddr *net.UDP
 		return
 	}
 
+	// If there are destination ips that we don't want to route, we
+	// have to do a dns lookup here to find the destination ip.
+	if c.ignoreDsts != nil {
+		if len(msg.Questions) > 0 {
+			q := msg.Questions[0]
+			switch q.Type {
+			case dnsmessage.TypeAAAA, dnsmessage.TypeA:
+				dstAddrs, err := lookupDestinationIP(q.Name.String())
+				if err != nil {
+					log.Printf("HandleDNS: lookup destination failed: %v\n ", err)
+					return
+				}
+				if c.ignoreDestination(dstAddrs) {
+					bs, err := dnsResponse(&msg, dstAddrs)
+					if err != nil {
+						log.Printf("HandleDNS: generate ignore response failed: %v\n", err)
```

Comment:
in the error case we probably want to return early.

Reply:
when running locally i see messages like (I changed the log line to also include the type)

```
2024/06/05 09:59:27 HandleDNS: generate ignore response failed: "TypeAAAA" no address found
```

which aren't really useful